banyand/measure/merger.go

```diff
@@ -179,15 +196,23 @@ var reservedDiskSpace uint64
 
 func (tst *tsTable) getPartsToMerge(snapshot *snapshot, maxFanOut uint64) ([]*partWrapper, map[uint64]struct{}) {
 	var parts []*partWrapper
-	toBeMerged := make(map[uint64]struct{})
+
 	for _, pw := range snapshot.parts {
-		if pw.mp != nil || pw.p.partMetadata.TotalCount < 1 || pw.p.partMetadata.CompressedSizeBytes > maxFanOut {
+		if pw.mp != nil || pw.p.partMetadata.TotalCount < 1 {
 			continue
 		}
 		parts = append(parts, pw)
+	}
+
+	tst.pwsChunk = mergerPolicy.getPartsToMerge(tst.pwsChunk[:0], parts, maxFanOut)
```

Comment:
Could you use a local var instead of a field?

Reply:
I think this could help reduce memory alloc

Reply:
The given slice consists of pointers. It is feasible to allocate it on the stack as the process is quick enough.